### PR TITLE
feat: add retry and check binary

### PR DESCRIPTION
**Pull request by purelind**
add retry and check binary.

### CODE REVIEW

**Comment by purelind** [comment]
retest link
- [ ] https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap-qe%2Ftidb-test%2Fghpr_integration_mysql_test/detail/ghpr_integration_mysql_test/294/pipeline

File: pipelines/PingCAP-QE/tidb-test/latest/ghpr_common_test.groovy [hunk 1]
```diff
@@ -67,13 +67,21 @@ pipeline {
...
                 dir('tidb') {
                     cache(path: "./bin", includes: '**/*', key: "ws/${BUILD_TAG}/dependencies") {
-                        sh label: 'tidb-server', script: 'ls bin/tidb-server || make'
-                        sh label: 'download binary', script: """
-                            chmod +x ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/*.sh
-                            ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/download_pingcap_artifact.sh --pd=${REFS.base_ref} --tikv=${REFS.base_ref}
-                            mv third_bin/* bin/
-                            ls -alh bin/
-                        """
+                        sh label: 'tidb-server', script: 'make'
```
Comment (wuhuizuo) [inline pipelines/PingCAP-QE/tidb-test/latest/ghpr_common_test.groovy:70]: why not detect the cache first for tidb-server?

File: pipelines/PingCAP-QE/tidb-test/latest/ghpr_integration_python_orm_test.groovy [hunk 2]
```diff
@@ -67,13 +67,22 @@ pipeline {
...
+                        retry(2) {
+                            sh label: 'download binary', script: """
+                                chmod +x ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/*.sh
+                                ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/download_pingcap_artifact.sh --pd=${REFS.base_ref} --tikv=${REFS.base_ref}
+                                mv third_bin/* bin/
+                                ls -alh bin/
+                            """
+                            sh label: "check binary", script: """
+                                pwd && ls -alh
+                                ls bin/tidb-server && chmod +x bin/tidb-server && ./bin/tidb-server -V
```
Comment (wuhuizuo) [inline pipelines/PingCAP-QE/tidb-test/latest/ghpr_integration_python_orm_test.groovy:80]: can we push the chmod step into the download_pingcap_artifact.sh file?

File: pipelines/PingCAP-QE/tidb-test/latest/ghpr_common_test.groovy [hunk 3]
```diff
@@ -67,13 +67,21 @@ pipeline {
...
+                        retry(2) {
+                            sh label: 'download binary', script: """
+                                chmod +x ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/*.sh
+                                ${WORKSPACE}/scripts/PingCAP-QE/tidb-test/download_pingcap_artifact.sh --pd=${REFS.base_ref} --tikv=${REFS.base_ref}
```
Comment (wuhuizuo) [inline pipelines/PingCAP-QE/tidb-test/latest/ghpr_common_test.groovy:74]: can we implement the retry in `download_pingcap_artifact.sh` script?

**Pull request merged**